File: jogodaforca/jogodaforcav3.py
import requests
import warnings
from lxml import html
import random
from os import system, name

class Enforcado:

    def __init__(self):
        self.tabuleiro = []
        self.palavra = ""
        self.lista_letras_frases = []
        
    def limpar_tela(self):
        if name == 'nt':
            _ = system('cls')
        else:
            _ = system('clear')
        print("\nBem-vindo(a) ao jogo da forca com o tema filmes!")
        print("Adivinhe o nome do filme abaixo:\n")            
        
    def mostrar_enforcado(self,chances):
        board = ['''

>>>>>>>>>>Enforcado<<<<<<<<<<

+---+
|   |
    |
    |
    |
    |
=========''', '''

+---+
|   |
O   |
    |
    |
    |
=========''', '''

+---+
|   |
O   |
|   |
    |
    |
=========''', '''

 +---+
 |   |
 O   |
/|   |
     |
     |
=========''', '''

 +---+
 |   |
 O   |
/|\  |
     |
     |
=========''', '''

 +---+
 |   |
 O   |
/|\  |
/    |
     |
=========''', '''

 +---+
 |   |
 O   |
/|\  |
/ \  |
     |
=========''']

        return board[chances]
   
    def obter_frases(self):
        with warnings.catch_warnings():
            warnings.simplefilter("ignore")
            response = requests.get('https://www.adorocinema.com/filmes-todos/',verify=False)
        try:
            content = response.content
            tree = html.fromstring(content)
            movielist = tree.xpath('//h2[@class="meta-title"]/a')
            return [element.text for element in movielist]
        except:
            return ['A volta dos que não foram','Poeira em alto mar','As tranças do vovô careca']   

    def adivinhar_letra(self):
        chances = 6

        letras_tentativas = []

        while chances > 0:

            self.mostrar_letra_no_board(6 - chances)

            tentativa = input("\nDigite uma letra: ")

            if tentativa in letras_tentativas:
                print("Você já tentou essa letra. Escolha outra!")
                continue

            letras_tentativas.append(tentativa)

            if tentativa.capitalize() in self.lista_letras_frases:

                print("Você acertou a letra!")

                for indice in range(len(self.lista_letras_frases)):

                    if self.lista_letras_frases[indice].capitalize() == tentativa.capitalize():
                        self.tabuleiro[indice] = tentativa

                if self.venceu():
                    break
            else:
                print("Ops. Essa letra não está no nome do filme!")
                chances -= 1
	
    def venceu(self):
        if "_" not in self.tabuleiro:
            print("\nVocê venceu! O nome do filme era: {}".format(self.palavra))
            return True
        return False
        
    def mostrar_letra_no_board(self,chances):
        print(self.mostrar_enforcado(chances))
        print("Nome do filme: ", self.tabuleiro)
        print("\n")
		
    def imprimir_board(self):
        self.limpar_tela()

        frases = self.obter_frases()

        self.palavra = random.choice(frases)

        self.lista_letras_frases = [letra.capitalize() for letra in self.palavra]

        self.tabuleiro = ["_"] * len(self.palavra)

        self.adivinhar_letra()

        if "_" in self.tabuleiro:
            print("\nVocê perdeu! O nome do filme era: {}.".format(self.palavra))           
         
if __name__ == "__main__":
    enforcado = Enforcado()
    enforcado.imprimir_board()

File: jogodaforca/jogodaforcav3.php
<?php
error_reporting(E_ERROR | E_PARSE);

class Enforcado {
    private array $tabuleiro;
    private string $palavra;
    private array $lista_letras_frases;

    public function __construct()
    {
        $this->tabuleiro = [];
        $this->palavra = "";
        $this->lista_letras_frases = [];
    }
        
    public function limpar_tela()
    {
        system(str_contains(PHP_OS,'NT') ? 'cls' :'clear');
        echo "\nBem-vindo(a) ao jogo da forca com o tema filmes!\n";
        echo "Adivinhe o nome do filme abaixo:\n";
    }                    
        
    public function mostrar_enforcado($chances)
    {
        $board = [
<<<BLOCO
=========Enforcado==========

+---+
|   |
    |
    |
    |
    |
=========
BLOCO,
<<<BLOCO

+---+
|   |
O   |
    |
    |
    |
=========
BLOCO,
<<<BLOCO

+---+
|   |
O   |
|   |
    |
    |
=========
BLOCO,
<<<BLOCO

 +---+
 |   |
 O   |
/|   |
     |
     |
=========
BLOCO,
<<<BLOCO

 +---+
 |   |
 O   |
/|\  |
     |
     |
=========
BLOCO,
<<<BLOCO

 +---+
 |   |
 O   |
/|\  |
/    |
     |
=========
BLOCO,
<<<BLOCO

 +---+
 |   |
 O   |
/|\  |
/ \  |
     |
=========
BLOCO
        ];
        return $board[$chances];
    }
   
    public function obter_frases()
    {
        @$response = file_get_contents('https://www.adorocinema.com/filmes-todos/');
        try {
            $doc = new DOMDocument();
            $doc->loadHTML($response);
            $xpath = new DOMXpath($doc);
            $elements = $xpath->query('//h2[@class="meta-title"]/a');
            $movielist = [];
            foreach($elements as $element)
            {
                $movielist[] = $element->nodeValue;
            }
            return $movielist;
            
        } catch (Exception $e) {
            return ['A volta dos que não foram','Poeira em alto mar','As tranças do vovô careca'];
        }
    }
    
    public function adivinhar_letra()
    {
        $chances = 6;

        $letras_tentativas = [];

        while ($chances > 0){

            $this->mostrar_letra_no_board(6 - $chances);

            echo "\nDigite uma letra: ";
            $tentativa = readline();

            if (in_array($tentativa, $letras_tentativas)){
                echo "Você já tentou essa letra. Escolha outra!";
                continue;
            }

            $letras_tentativas[] = $tentativa;

            if (in_array(strtoupper($tentativa), $this->lista_letras_frases)) {

                echo "\nVocê acertou a letra!\n";

                foreach($this->lista_letras_frases as $indice => $letra){

                    if (strtoupper($letra) == strtoupper($tentativa)){
                        $this->tabuleiro[$indice] = $tentativa;
                    }
                }

                if ($this->venceu()) {
                    break;
                }
            } else {
                echo "\nOps. Essa letra não está no nome do filme!\n";
                $chances -= 1;
            }
        }
    }
	
    public function venceu()
    {
        if (!in_array("_",$this->tabuleiro))
        {
            echo "\nVocê venceu! O nome do filme era: {$this->palavra}\n";
            return true;
        }
        return false;
    }
        
    public function mostrar_letra_no_board($chances)
    {
        echo $this->mostrar_enforcado($chances);
        echo "\nNome do filme: " . implode(' ',$this->tabuleiro) . "\n";
    }
		
    public function imprimir_board()
    {
        $this->limpar_tela();

        $frases = $this->obter_frases();

        $this->palavra = $frases[array_rand($frases,1)];

        for($i=0;$i<strlen($this->palavra);$i++){
            $this->lista_letras_frases[] = strtoupper($this->palavra[$i]);
        }

        $this->tabuleiro = array_fill(0,strlen($this->palavra),"_");

        $this->adivinhar_letra();

        if (in_array("_", $this->tabuleiro)){
            echo "\nVocê perdeu! O nome do filme era: {$this->palavra}\n";
        }
    }
}

$enforcado = new Enforcado();
$enforcado->imprimir_board();
